web/src/components/storage/iscsi/TargetsSection.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useState } from "react";
+import { Button, Toolbar, ToolbarItem, ToolbarContent, Stack } from "@patternfly/react-core";
+import { Section } from "~/components/core";
+import { NodesPresenter, DiscoverForm } from "~/components/storage/iscsi";
+import { _ } from "~/i18n";
+import { useNodes, useNodesChanges } from "~/queries/storage/iscsi";
+import { discover } from "~/api/storage/iscsi";
+
+export default function TargetsSection() {
+  const [isDiscoverFormOpen, setIsDiscoverFormOpen] = useState<boolean>(false);
+  const nodes = useNodes();
+  useNodesChanges();
```

Comment:
Is there some changes done in `changes` queries? Previously it returns query. So it can be written like `const nodes = useNodesChanges();` . And it is still done this way e.g. in DASD, but in others where it was before I do not see it anymore. ( e.g. https://github.com/openSUSE/agama/blob/master/web/src/queries/dasd.ts#L121 )

Reply:
Well, that's an interesting topic. We usually have separate hooks for retrieving the query and subscribing to changes. DASD handling is the only place where is done differently. DASD hooks 1) retrieve the value and 2) subscribe to the changes. So, if you use the `useDASDFormatJobsChanges` in more than one place, you are subscribing twice. In this particular case, the `JobAdded` event will not be handled correctly (that's easy to fix).

Let's consider that our DASD formats progress is not blocking (we discussed that during the review), and you want to add a counter to the pending jobs. Then you might want to access that query from your counter, too. And then, you will find the problem mentioned above.

Having two different hooks (`useIssues` and `useIssuesChanges`) might be a problem too. You need to be sure that the changes hooks are used at least at the same level as the query hook. That's why most `changes` hooks are called from the `App` component.

Finally, something we need to fix, is adapting all those *changes* queries to not update the query if it is not defined. I did that correctly (I think) in this pull request, but we might need to fix others.

Reply:
Perhaps I should move the call to `useNodesChanges` to the `App` level.